### DISCUSSION TITLE
Support for ARM64?

### DISCUSSION THREAD

**Discussion by root9191** [JulianPrieber/littlelink-custom#100] [General]
Hello, 
will there be coming a Docker Image for arm64 too or just amd?

**Comment by JulianPrieber** [comment]
**This is something I will be definitely looking into!**

**Comment by JulianPrieber** [comment]
I threw this thing together real quick: https://hub.docker.com/r/julianprieber/llc-arm.

Not sure if this will work at all. At this moment, I have nothing to test this with.

**Reply by root9191** [reply]
So i just tried this on my Raspberry Pi and what should i say: It worked like a charm :)
Thank you, for your fast Support.

**Reply by JulianPrieber** [reply]
Alright. 👍 Always glad I can help.

I will turn this into a full release.

**Reply by root9191** [reply]
Thanks a lot. The only problem i have for now is, that i cannot change the Site Logo, only the Page Logo.

**Reply by JulianPrieber** [reply]
Might be a caching issue that could be solved by pressing `ctrl + f5`. Otherwise, I'm not too sure, might be a permissions issue or something with the docker.

**Reply by root9191** [reply]
Hmm OK. So for the Profile its working, that's enogh for me.
Does the backup also include the Profile Settings and the Links? Or just the .env File?

**Reply by JulianPrieber** [reply]
The backup includes all files, including every setting you make. You can download your backup from the backups tab in your config.

Should you ever want to migrate your instance to the web version, you can just use your downloaded backup for that. The two versions don't differ in that regard.

**Reply by root9191** [reply]
Hmm then that's maybe another issue in the arm Version. If i download the Backup i only get the .env File.
Custom Themes are also not working.

**Reply by JulianPrieber** [reply]
Click on "You can find your updater backups [here](#)"

> Custom Themes are also not working.

That really sounds like a caching issue, are accessing your instance directly or through a proxy?

**Reply by root9191** [reply]
This is empty when i click on my Backups.
Yep i use the Nginx Proxy Manager.
is there some documentation for this?

**Reply by JulianPrieber** [reply]
Your updater backups are empty until you make an update. Only the updater takes backups.
 
I had some issues before using NGINX as a proxy, granted I'm not really experienced in that regard.
<br>

> is there some documentation for this?

If you are referring to documentation about Littlelink Custom and proxies, we currently have nothing in that direction.
Should the problem really be the reverse proxy, you can easily test this by accessing your docker locally over the local IP.

If the proxy causes this, you might get some help at the [DB tech Discord](https://dbt3.ch/going/5/https://discord.gg/M9J6hFq) or perhaps [our community Discord](https://discord.littlelink-custom.com).

**Reply by root9191** [reply]
OK, maybe i'll give this a try. 
Is there a possibilty to disable the Registration from the Home Menu?

**Reply by JulianPrieber** [reply]
It's on the pages section on the Admin Panel
![chrome_bhYlClcKW3](https://user-images.githubusercontent.com/60265788/179027991-cb35b61e-dc6a-40d1-8ee7-61602d77242a.png)

**Reply by root9191** [reply]
Aah there it is :D
Thanks a lot!

**Reply by root9191** [reply]
Is it possible to remove the Automatic URL, when i use an Email so that just stands there: mailto:admin@adress.com
![Bildschirmfoto 2022-07-16 um 18 39 23](https://user-images.githubusercontent.com/47309435/179364124-32c837c6-1ccc-4f74-9726-b9439dd5161a.png)
?

**Reply by JulianPrieber** [reply]
I'm currently working on that system. I have something like this on the beta server that might work for you.

You can just enter mails without the mailto in the mail link, and it should just work.

You can follow step 2 in this guide if you don't know how to install beta versions: https://littlelink-custom.com/beta-tester#step-2

**Reply by root9191** [reply]
OK i use the Beta. I also have a Backup now, cause of the Update.
Would be cool, to do that manually too. I tried to edit my Link, but after saving i always get "https://" before the Mail Address.
So i added a new one and that worked so far, but the Link generated is the Same as in the official Release. 
I just don't want opening a new Tab for this.

**Reply by root9191** [reply]
Is it possible, that something is wrong with your custom Themes? Cause when i look into the preview.png of Sublime and Flare, the Preview Files look the same:
![Bildschirmfoto 2022-07-19 um 20 28 18](https://user-images.githubusercontent.com/47309435/179823078-544b7d96-2c04-48b9-b4c3-6de6c8120743.png)

**Reply by JulianPrieber** [reply]
Probably just a caching issue. I'd recommend turning off any cache.

**Reply by root9191** [reply]
i only downloaded this Themes and compared the Preview and the Size of the Zip File and they are the same? 
Edit:
Here is the Readme of Flare:
![image](https://user-images.githubusercontent.com/47309435/179850488-46d19e0c-ef15-4526-923c-7d671070f37c.png)

**Reply by JulianPrieber** [reply]
Dam, you're right. The downloaded theme files contain files from Galaxy.

The weir thing is the source files are completely normal, and contain the correct files.

**Reply by JulianPrieber** [reply]
It was the tags. Should be fixed now, thanks for pointing that out 👍.

**Reply by root9191** [reply]
Thank you. Really weird is, that the Themes are now working properly :)

**Reply by root9191** [reply]
btw: do you know, why the Docker Container runs unhealthy, when i set force_https to true?

**Reply by JulianPrieber** [reply]
This might be causing issues on the docker version when behind a wrongly configured reverse proxy, I think.

**Reply by root9191** [reply]
Hmm OK, sadly i didn't get much response on Discord for this Topic.
Is there a way to manually backup Files through the Docker Container? Through CLI maybe?

**Reply by JulianPrieber** [reply]
I'd love to help, but I'm just not experienced enough to effectively troubleshoot :(. I can mostly resolve any issues by googling.

For the docker backups, there's not really a need for backups besides from updates. You can just back up the docker image itself. 
The best way to back up your instance is probably turning on bet mode and updating normally. Depending on the current beta status, I cannot recommend this, tho.

There is a command line command to do the backup manually.

**Reply by JulianPrieber** [reply]
The command for doing  backups should be somewhere on here: https://github.com/spatie/laravel-backup/blob/main/README.md

**Reply by JulianPrieber** [reply]
Something like:
```
cd /htdocs
php artisan backup:run
```

**Reply by root9191** [reply]
Thanks a lot.
The Backup first failed becuase SQLITE was not installed:
`Backup failed because: The dump process failed with exitcode 127 : Command not found : sh: sqlite3: not found`
After installing, the Backup completes successfully and i can see the Backup in the Webinterface 👍

**Reply by JulianPrieber** [reply]
`php artisan backup:run --only-files`

Should always work.

**Reply by root9191** [reply]
Is  there a way to configure something in Apache2? I wanna try something i found by Google, but that needs the a2enmod Command in the CLI but the command  was not found. I haven't located it somewehere, could it be something to do with the $PATH?

**Reply by JulianPrieber** [reply]
On the built-in phpinfo page, you can see all your installed Apache modules. Everything else you'd have to find out how to do. Most things are at stock setting if they're not required to run LittleLink Custom.

You should be able to easily find guides online by searching for what you want to do, followed by the OS name "Alpine Linux".
